src/homeform.cpp

```diff
@@ -1096,782 +1297,827 @@ void homeform::update()
-    {
-        if(b.name().trimmed().length())
+    r.append(QStringLiteral("Disabled"));
+    for (const QBluetoothDeviceInfo &b : qAsConst(bluetoothManager->devices)) {
+        if (!b.name().trimmed().isEmpty()) {
+
             r.append(b.name());
+        }
     }
     return r;
 }
 
-QStringList homeform::metrics()
-{
-    return bluetoothdevice::metrics();
-}
+QStringList homeform::metrics() { return bluetoothdevice::metrics(); }
 
-struct OAuth2Parameter
-{
-    QString responseType = "code";
-    QString approval_prompt = "force";
+struct OAuth2Parameter {
+    QString responseType = QStringLiteral("code");
+    QString approval_prompt = QStringLiteral("force");
 
-    inline bool isEmpty() const{
-        return responseType.isEmpty() && approval_prompt.isEmpty();
-    }
-    QString toString() const{
+    inline bool isEmpty() const { return responseType.isEmpty() && approval_prompt.isEmpty(); }
+
+    QString toString() const {
         QString msg;
         QTextStream out(&msg);
-        out << "OAuth2Parameter{\n"
-            << "responseType: " << this->responseType << "\n"
-            << "approval_prompt: " << this->approval_prompt << "\n";
+        out << QStringLiteral("OAuth2Parameter{\n") << QStringLiteral("responseType: ") << this->responseType
+            << QStringLiteral("\n") << QStringLiteral("approval_prompt: ") << this->approval_prompt
+            << QStringLiteral("\n");
         return msg;
     }
 };
 
-QAbstractOAuth::ModifyParametersFunction homeform::buildModifyParametersFunction(QUrl clientIdentifier,QUrl clientIdentifierSharedKey){
-    return [clientIdentifier,clientIdentifierSharedKey]
-            (QAbstractOAuth::Stage stage, QVariantMap *parameters){
-        if(stage == QAbstractOAuth::Stage::RequestingAuthorization){
-            parameters->insert("responseType","code"); /* Request refresh token*/
-            parameters->insert("approval_prompt","force"); /* force user check scope again */
-            QByteArray code = parameters->value("code").toByteArray();
-            (*parameters)["code"] = QUrl::fromPercentEncoding(code);
+QAbstractOAuth::ModifyParametersFunction
+homeform::buildModifyParametersFunction(const QUrl &clientIdentifier, const QUrl &clientIdentifierSharedKey) {
+    return [clientIdentifier, clientIdentifierSharedKey](QAbstractOAuth::Stage stage, QVariantMap *parameters) {
+        if (stage == QAbstractOAuth::Stage::RequestingAuthorization) {
+            parameters->insert(QStringLiteral("responseType"), QStringLiteral("code")); /* Request refresh token*/
+            parameters->insert(QStringLiteral("approval_prompt"),
+                               QStringLiteral("force")); /* force user check scope again */
+            QByteArray code = parameters->value(QStringLiteral("code")).toByteArray();
+            // NOTE: Maybe this could be a better alternative
+            // parameters->find(QStringLiteral("code"))->setValue(QUrl::fromPercentEncoding(code));
```

Comment:
@p3g4asus what do you think about this modification that @lifof did?

Reply:
This is a good catch by @lifof because, looking at [qt source code](https://code.woboq.org/qt5/qtnetworkauth/src/oauth/qoauth2authorizationcodeflow.cpp.html#392), "code" is stored inside parameters `QVariaintMap` percent-encoded. So I think you should accept its advice! Good work!

Reply:
@lifof could you try your solution? Looks good to me. You have to disconnect your account from QZ and then reconnect it in order to test this

Reply:
This modification seems to be just fine to me. It is probably irrelevant but @lilof solution seems to be more correct to me.

Reply:
@cagnulein @p3g4asus 
The static analyzer did it's job, actually I'm still circling around the code base from time to time as It seems to be quite huge, especially that homeform.cpp , testing this is not possible for me at the moment as I'm still trying to digest the rest of the code :sweat_smile: